src/graphql/client_graph_node.rs

```diff
@@ -44,11 +57,97 @@ pub async fn query_graph_node_network_block_hash(
     if let Some(data) = response_body.data {
         match data.block_hash_from_number {
             Some(hash) => Ok(hash),
-            None => Err(QueryError::EmptyResponseError(
+            None => Err(QueryError::ParseResponseError(
                 "No block hash from number".to_string(),
             )),
         }
     } else {
-        Err(QueryError::EmptyResponseError(graph_node_endpoint))
+        Err(QueryError::ParseResponseError(format!(
+            "Network {network} blockhash at block {block_number}"
+        )))
     }
 }
+
+/// Query graph node for Indexing Statuses
+pub async fn perform_indexing_statuses(
+    graph_node_endpoint: String,
+    variables: indexing_statuses::Variables,
+) -> Result<reqwest::Response, reqwest::Error> {
+    let request_body = IndexingStatuses::build_query(variables);
+    let client = reqwest::Client::new();
+    client
+        .post(graph_node_endpoint)
+        .json(&request_body)
+        .send()
+        .await?
+        .error_for_status()
+}
+
+/// Construct GraphQL variables and parse result for Proof of Indexing.
+/// For other radio use cases, provide a function that returns a string
+pub async fn update_network_chainheads(
+    graph_node_endpoint: String,
+    network_map: &mut HashMap<NetworkName, BlockPointer>,
+) -> Result<HashMap<String, NetworkPointer>, QueryError> {
+    let variables: indexing_statuses::Variables = indexing_statuses::Variables {};
+    let queried_result = perform_indexing_statuses(graph_node_endpoint.clone(), variables).await?;
+    let response_body: Response<indexing_statuses::ResponseData> = queried_result.json().await?;
+
+    // subgraph (network, latest blocks)
+    let mut subgraph_network_blocks: HashMap<String, NetworkPointer> = HashMap::new();
+
+    let updated_networks = response_body
+        .data
+        .map(|data| {
+            data.indexing_statuses
+                .into_iter()
+                .map(|status| {
+                    status
+                        .chains
+                        .into_iter()
+                        .map(|chain| {
+                            let network_name = chain.network.clone();
+                            let _chainhead_block = chain
```

Comment:
should we remove unused vars?

Reply:
This was intentional, we do need this block to be eagerly evaluated to update the networks' block pointers. This was the compiler's suggested way for evaluating this block, open to suggestions on a different style

Reply:
updated to the refactored version and the unused var has been removed